R/pretty_lollipop_plot.R

```diff
@@ -184,34 +332,114 @@ pretty_lollipop_plot <- function(
             data = domain_data, 
             aes(
                 x = text.position, 
-                y = 0, 
+                y = 0.1, 
                 label = text.label
                 )
-        ) +
-        labs(
-            x = "AA Position", 
-            y = "Mutation Count", 
-            title = paste0(
-                plot_title, 
-                "\n[Somatic Mutation Rate: ", 
-                Somatic_Mutation_Rate, 
-                "%]"
+        ) 
+
+        if (mirrorarg == TRUE) {
+            plot <- plot +
+                annotate(
+                    "text",
+                    x = 0,
+                    y = max(gene_counts$mutation_count) * 1.1,
+                    label = paste0(
+                        "Somatic Mutation Rate ", 
+                        Somatic_Mutation_Rate[1], 
+                        "% N = ", 
+                        Somatic_Mutation_Denominator[1],
+                        "\n",
+                        "Comparison Value ", 
+                        '"', 
+                        Sample1, 
+                        '"'
+                    ),
+                    hjust = 0
+                ) +
+                annotate(
+                    "text",
+                    x = 0,
+                    y = min(gene_counts$mutation_count) * 1.1,
+                    label = paste0(
+                        "Somatic Mutation Rate ", 
+                        Somatic_Mutation_Rate[2], 
+                        "% N = ",
+                        Somatic_Mutation_Denominator[2], 
+                        "\n", 
+                        "Comparison Value ",
+                        '"', 
+                        Sample2, 
+                        '"'
+                    ),
+                    hjust = 0
+                ) +
+                geom_text(
+                    data = domain_data, 
+                    aes(
+                        x = text.position, 
+                        y = -0.2, 
+                        label = paste0(
+                            "p = ", 
+                            round(p_value, 3)
+                        )  # Display domain-specific p-value
+                    )
+                ) +
+                labs(
+                    x = "AA Position", 
+                    y = "Mutation Count", 
+                    title = paste0(
+                        plot_title,
+                        "\n",
+                        "p = ",
+                        round(gene_p_value, 3)
+                    )
                 )
-        ) +
+        } else {
+            plot <- plot +
+                annotate(
+                    "text",
+                    x = 0,
+                    y = max(gene_counts$mutation_count) * 1.1,
+                    label = paste0(
```

Comment:
I was going to remove this feature from the version I have been working on. I don't think it's wise to calculate the % mutated from the maf file - mafs are frequently subset by gene and/or patient ID and the maf probably doesn't represent the true denominator in many cases.